src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/OrderedDictionary.cs

```diff
@@ -529,14 +529,14 @@ public void RemoveAt(int index)
             RemoveEntryFromBucket(index);
 
             // Decrement the indices > index
+            Entry[] entries = _entries;
             for (int i = index + 1; i < count; ++i)
             {
+                entries[i - 1] = entries[i];
                 UpdateBucketIndex(i, -1);
```

Comment:
At a guess for large enough Array.Copy is probably always going to be faster than looping, because it boils down to "memcpy". Can you explain the purpose of the change here? I'm missing it.

Reply:
Oh, I see. BEcause of the chaining.